File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string a, string b)
{
    return a+b > b+a;
}

string solution(vector<int> numbers) {
    string answer ="";
    
    vector<string> nums;
    
    for(int& item : numbers)
    {
        nums.push_back(to_string(item));
    }
    
    sort(nums.begin(), nums.end(), compare);
    
    for(string& item : nums)
    {
        answer += item;
    }
    
    if(answer[0] == '0') return "0";
    
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    var answer = '';
    
    nums = numbers.sort((a,b) => {
        str1 = a.toString() + b.toString();
        str2 = b.toString() + a.toString();
        if(str1<str2) return 1;
        else if (str1>str2) return -1;
        else 0;
    });
    
    answer = nums.reduce((acc, idx) => acc + idx,'')
    
    if(parseInt(answer) === 0) return '0';
    
    return answer;
}